app/integration_action.go

```diff
@@ -250,6 +257,87 @@ func (a *App) DoActionRequest(rawURL string, body []byte) (*http.Response, *mode
 	return resp, nil
 }
 
+type LocalResponseWriter struct {
+	data   []byte
+	status int
+}
+
+func (w *LocalResponseWriter) Header() http.Header {
+	return make(http.Header)
+}
+
+func (w *LocalResponseWriter) Write(bytes []byte) (int, error) {
+	w.data = make([]byte, len(bytes))
+	copy(w.data, bytes)
+	return len(w.data), nil
+}
+
+func (w *LocalResponseWriter) WriteHeader(statusCode int) {
+	w.status = statusCode
+}
+
+func (a *App) DoLocalRequest(rawURL string, body []byte) (*http.Response, *model.AppError) {
+	inURL, err := url.Parse(rawURL)
+	if err != nil {
+		return nil, model.NewAppError("DoActionRequest", "api.post.do_action.action_integration.app_error", nil, "err="+err.Error(), http.StatusBadRequest)
+	}
+	pluginsEnvironment := a.GetPluginsEnvironment()
+
+	regx := regexp.MustCompile(`plugins\/\s*([^\/]*)`)
```

Comment:
Regular expression parsing here seems overkill. Can we just `strings.Split` on `/` and check that `result[0] == "plugins"` and then verify that `result[1]` is a plugin that exists? We can then remove that prefix from the URL and use it directly below.

Reply:
In relative urls, there could be a leading '/' so a simple split has two cases to handle.  I originally did this and then opted for a regex.  I will use a strings.Split as requested.